hack/docgen/main.go

```diff
@@ -163,56 +249,84 @@ func parseFieldType(p interface{}) string {
 		return parseFieldType(t.Sel)
 	default:
 		log.Printf("unknown: %#v", t)
+
 		return ""
 	}
 }
 
-func collectFields(s *structType) (entries []*Entry) {
-	entries = []*Entry{}
+func escape(value string) string {
+	return strings.TrimSpace(strings.ReplaceAll(
+		strings.ReplaceAll(value, "\"", "\\\""),
+		"\n",
+		"\\n",
+	))
+}
+
+func collectFields(s *structType) (fields []*Field) {
+	fields = []*Field{}
 
-	for _, field := range s.node.Fields.List {
-		if field.Tag == nil {
-			if field.Names == nil {
+	for _, f := range s.node.Fields.List {
+		if f.Tag == nil {
+			if f.Names == nil {
 				// This is an embedded struct.
 				continue
 			}
-			log.Fatalf("field %q is missing a yaml tag", field.Names[0].Name)
 		}
 
-		if field.Doc == nil {
-			log.Fatalf("field %q is missing a documentation", field.Names[0].Name)
+		if f.Doc == nil {
+			log.Fatalf("field %q is missing a documentation", f.Names[0].Name)
 		}
 
-		tag := reflect.StructTag(strings.Trim(field.Tag.Value, "`"))
-		name := tag.Get("yaml")
-		name = strings.Split(name, ",")[0]
+		name := f.Names[0].Name
+
+		fieldType := parseFieldType(f.Type)
 
-		fieldType := parseFieldType(field.Type)
+		tag := reflect.StructTag(strings.Trim(f.Tag.Value, "`"))
+		yamlTag := tag.Get("yaml")
+		yamlTag = strings.Split(yamlTag, ",")[0]
 
-		text := &Text{}
-		if err := yaml.Unmarshal([]byte(field.Doc.Text()), text); err != nil {
-			log.Fatal(err)
+		if yamlTag == "" {
+			yamlTag = strings.ToLower(yamlTag)
 		}
 
-		entry := &Entry{
+		text := parseComment([]byte(f.Doc.Text()))
+
+		field := &Field{
 			Name: name,
+			Tag:  yamlTag,
 			Type: fieldType,
 			Text: text,
 		}
 
-		if field.Comment != nil {
-			entry.Note = field.Comment.Text()
+		if f.Comment != nil {
+			field.Note = escape(f.Comment.Text())
 		}
 
-		entries = append(entries, entry)
+		fields = append(fields, field)
 	}
 
-	return entries
+	return fields
 }
 
-func render(section *Section, f *os.File) {
-	t := template.Must(template.New("section.tpl").Parse(tpl))
-	err := t.Execute(f, section)
+func render(doc *Doc, dest string) {
+	t := template.Must(template.New("docfile.tpl").Parse(tpl))
+	buf := bytes.Buffer{}
+
+	err := t.Execute(&buf, doc)
+	if err != nil {
+		panic(err)
+	}
+
+	formatted, err := format.Source(buf.Bytes(), format.Options{})
```

Comment:
I guess we need to add our options for the current module

Reply:
we don't really rely on any internal packages there, so I guess it's fine to skip adding any options. And linter seems to be happy anyways